### PR TITLE
feat: add button component and styles

### PR DESCRIPTION
**Pull request by trinegl**

### CODE REVIEW

File: src/styles/core/_base.scss [hunk 1]
```diff
@@ -22,6 +22,15 @@ body {
 	}
 }
 
+// Remove outline by default, is added with JavaScript the first time user hits "tab"
```
Comment (dariobk) [inline src/styles/core/_base.scss:25]: I just tried `	&:not(:focus-visible) {
		outline: none;
	}` for the icon-button and that seems to work. Maybe that could work here too 🧐

Comment (s2-schwaddi) [reply src/styles/core/_base.scss:25]: +1 for that.

`:focus-visible` will only let outline appear on tab focus

File: src/styles/core/_button.scss [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+			stroke: $color-primary--black;
+		}
+	}
+
+	&--secondary {
+		background-color: $color-primary--beige;
+		color: $color-primary--black;
+
+		.is-dark & {
+			background-color: #181818;
```
Comment (s2-schwaddi) [inline src/styles/core/_button.scss:49]: please use the variable 🙏

File: src/styles/core/_button.scss [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+	@include interactive-100();
+
+	line-height: initial;
+	border: 0;
+	display: inline-flex;
+	align-items: center;
+
+	&__icon {
+		fill: none;
+		width: 1.6rem;
```
Comment (s2-schwaddi) [inline src/styles/core/_button.scss:13]: please check if we have a variable for that

File: src/styles/core/_button.scss [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+		.is-dark & {
+			color: $color-primary--white;
+		}
+	}
+
+	// buttonSize
+	&--large {
+		padding: $spacing--200 $spacing--500;
+		border-radius: 2.4rem;
```
Comment (s2-schwaddi) [inline src/styles/core/_button.scss:81]: do we have a variable for that?

File: src/styles/core/_button.scss [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+	// state
+	&:disabled {
+		color: #798086;
+
+		.button__text {
+			border-bottom: 0;
+		}
+
+		.button__icon {
+			stroke: #798086;
```
Comment (s2-schwaddi) [inline src/styles/core/_button.scss:103]: variable?

File: src/styles/core/_base.scss (hunk 1 above)
Comment (trinegl) [reply src/styles/core/_base.scss:25]: True, but `:focus-visible` is not supported in Safari + some other browsers. I can try to implement this though, and ask if design is okay with the preview.

File: src/styles/core/_button.scss (hunk 4 above)
Comment (trinegl) [reply src/styles/core/_button.scss:81]: No variable exists that would make sense here. Should I create new ones? As far as I can see the border-radius would only need adjustment if the font-size increases too much

File: src/_includes/assets/svg/icons.svg [hunk 6]
```diff
@@ -0,0 +1,8 @@
+<svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg">
+  <defs>
+    <g id="arrow-right">
+      <path d="M9 13L12.9393 9.06066C13.5251 8.47487 13.5251 7.52513 12.9393 6.93934L9 3" stroke-width="1.5"/>
+      <path d="M2 8H13" stroke-width="1.5"/>
+    <g>
```
Comment (timbru31) [inline src/_includes/assets/svg/icons.svg:6]:
```suggestion
    </g>
```

**Review by Wandkleister — APPROVED** [review]

**Pull request merged**